src/main/java/com/minecolonies/blocks/AbstractBlockMineColonies.java

```diff
@@ -0,0 +1,64 @@
+package com.minecolonies.blocks;
+
+import com.minecolonies.achievements.ModAchievements;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.material.MapColor;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.BlockPos;
+import net.minecraft.world.World;
+
+/**
+ * Abstract {@link Block} wrapper.
+ * 
+ * @author Isfirs
+ * @since 0.1
+ */
+public abstract class AbstractBlockMineColonies extends Block
+{
+
+    /**
+     * Parent constructor.
+     * 
+     * @param materialIn
+     * @see Block#Block(Material)
+     */
+    public AbstractBlockMineColonies(Material materialIn)
+    {
+        super(materialIn);
+    }
+
+    /**
+     * Parent constructor.
+     *
+     * @param blockMaterialIn
+     * @param blockMapColorIn
+     * @see Block#Block(Material, MapColor)
+     */
+    public AbstractBlockMineColonies(Material blockMaterialIn, MapColor blockMapColorIn)
+    {
+        super(blockMaterialIn, blockMapColorIn);
+    }
+
+    @Override
+    public void onBlockPlacedBy(World worldIn, BlockPos pos, IBlockState state, EntityLivingBase placer, ItemStack stack)
+    {
+        if (placer instanceof EntityPlayer)

```

Comment:
A few things here.
1) I realize your intention and why you created this abstract class, but we don't need it now, and if we need it in the future, we can create it. (I feel the same way about the Achievement and AchievementPage, but I'll let that one slide).
2) Currently the only block you check for is the townhall block, so lets just move this into the BlockHutTownhall class (It will also simplify the logic).
```
if (placer instanceof EntityPlayer)
{
    ((EntityPlayer) placer).triggerAchievement(ModAchievements.achievementBuildingTownhall);
}
```
Side note to moving this to the townhall: your argument is to put all the block placement achievements together is one spot, which makes sense if you know where they are, but you asked me where the townhall placement achievement was fired off, the first place I would guess would be in the townhall block.

3) the call to the super method should be before the achievement code.

4) We use vanilla code as a reference, but Mojang doesn't always follow the best programming practices.

5) achievementBuildingTownhall is the achievement for placing the townhall, but it sounds like it is for building it.